stdlib.sh

```diff
@@ -551,17 +551,26 @@ layout_pipenv() {
 # "$direnv_layout_dir/python-$python_version".
 #
 layout_pyenv() {
-  local python_version=$1
-  local pyenv_python
-  pyenv_python=$(pyenv root)/versions/${python_version}/bin/python
-  if [[ -x "$pyenv_python" ]]; then
-    if layout_python "$pyenv_python"; then
-      export PYENV_VERSION=$python_version
+  unset PYENV_VERSION
+  # layout_python prepends each python version to the PATH, so we add each
+  # version in reverse order so that the first listed version ends up
+  # first in the path
+  for ((j = $#; j >= 1; j--)); do
+    local python_version=${!j}
+    local pyenv_python
+    pyenv_python=$(pyenv root)/versions/${python_version}/bin/python
+    if [[ -x "$pyenv_python" ]]; then
+      if layout_python "$pyenv_python"; then
+        # e.g. Given "use pyenv 3.6.9 2.7.16", PYENV_VERSION becomes "3.6.9:2.7.16"
+        [[ -z "$PYENV_VERSION" ]] && PYENV_VERSION=$python_version || PYENV_VERSION="${python_version}:$PYENV_VERSION"
```

Comment:
There is a shorter notation that can be used here:
```
PYENV_VERSION=${python_version}${PYENV_VERSION:+:$PYENV_VERSION}
```

Reply:
Nice! I wasn't familiar with this, I had to look it up. For the curious: https://www.tldp.org/LDP/abs/html/parameter-substitution.html#PARAMALTV